empress/support_files/js/select-node-menu.js

```diff
@@ -45,73 +97,169 @@ define(["underscore", "util"], function (_, util) {
+     *                           internal node. Will be used in the diffLen
+     *                           message described above.
      */
-    SelectedNodeMenu.makeSampleMetadataTable = function (ctData, tableEle) {
-        tableEle.innerHTML = "";
-        // loop over all metadata fields the user has decided to show
-        var sortedFields = util.naturalSort(_.keys(ctData));
-        for (var i = 0; i < sortedFields.length; i++) {
-            var field = sortedFields[i];
-
-            // Create new rows in menu-table: the first row is for this
-            // metadata field's "headers" (the unique values in the field,
-            // e.g. "gut", "tongue", etc. for a field like body site), and
-            // the second row is for the sample presence data for
-            // the selected tree node within these unique values.
-            //
-            // Each group of two rows additionally has a header cell
-            // on its leftmost side which spans both the header and data
-            // rows; this header cell contains the name of the selected
-            // metadata field, and has some fancy CSS that keeps it frozen
-            // in place as the user scrolls the table horizontally.
-            var fieldHeaderRow = tableEle.insertRow(-1);
-            var fieldHeaderCell = fieldHeaderRow.insertCell(-1);
-            fieldHeaderCell.innerHTML = "<strong>" + field + "</strong>";
-            fieldHeaderCell.rowSpan = 2;
-            fieldHeaderCell.classList.add("menu-box-header-cell");
-            fieldHeaderCell.classList.add("frozen-cell");
-
-            var fieldDataRow = tableEle.insertRow(-1);
-
-            // add row values for this metadata field, one column at a time
-            var categories = util.naturalSort(_.keys(ctData[field]));
-            for (var j = 0; j < categories.length; j++) {
-                var categoryHeaderCell = fieldHeaderRow.insertCell(-1);
-                categoryHeaderCell.innerHTML =
-                    "<strong>" + categories[j] + "</strong>";
-                var categoryDataCell = fieldDataRow.insertCell(-1);
-                categoryDataCell.innerHTML = ctData[field][categories[j]];
+    SelectedNodeMenu.prototype.makeSampleMetadataTable = function (
+        ctData,
+        tipOrInt,
+        diffLen = 0,
+        descTipCt = 0
+    ) {
+        if (this.hasSampleMetadata) {
+            this.smTable.innerHTML = "";
+            if (_.isNull(ctData)) {
+                // This node (or its descendant tips) isn't present in the
+                // table. Just show some text explaining the situation, and
+                // hide the table.
+                var wtext;
+                if (tipOrInt === "tip") {
+                    wtext =
+                        "This is a tip in the tree. However, it is not " +
+                        "present in the input feature table, so we " +
+                        "can't show sample presence information for it.";
```

Comment:
What about a shorter version:

```suggestion
                        "This tree tip is not " +
                        "present in the input feature table, so we " +
                        "can't show sample presence information for it.";
```

Reply:
The reason I kept things like this was to be consistent with the other "notes" text that's shown in the alternative case (when this node is in the table) in Empress, [here](https://github.com/biocore/empress/blob/e982e65b5983299cddbf3665472bfd7d5cca254f/empress/support_files/js/select-node-menu.js#L283-L287) and [here](https://github.com/biocore/empress/blob/e982e65b5983299cddbf3665472bfd7d5cca254f/empress/support_files/js/select-node-menu.js#L389-L394).

I'd be down to change this in the future (maybe adding another UI element next to `Node length:` at the top of the menu that has something like `Node type:` and then `tip` or `internal`), but I would prefer to keep this as is for now. (I don't feel _super_ strongly about this, tho, so if you'd prefer we can change this up now :)